### ISSUE TITLE
Repeating data instances are joined by name 

### ISSUE THREAD

**Issue by martijnkersloot** [castoredc/castoRedc#25]
For `/study/{study_id}/data-point-collection/report-instance`, the R package joins repeating data instances by `report_instance_name`. This causes issues with multiple repeating data instances that have the same name, as it returns all those RDIs for all participant IDs, not the specific IDs.

**Steps to reproduce:**

- Make sure your study has duplicated repeating data instance names for different participant IDs, for example for participant 00001 - SAE 1 and for participant 00002 - SAE 1, 00003 - SAE 1 etc.
- Load the CastoRedc package following instructions [here](https://helpdesk.castoredc.com/link-castor-to-another-application/using-the-castor-api-r-package-for-data-analysis)
- Call the function: `reportdata <- castor_api$getReportInstancesBulk("study_id")`
- The function will return duplicated entries for the repeating data instances. It seems this is happening due to the R package joining the repeating data instances with by = join_by(report_instance_name) instead of unique ID: <img width="667" alt="7e6fc719-df84-4194-aaff-dc7c4022d586" src="https://github.com/castoredc/castoRedc/assets/2394455/9d07573e-b8ba-41e7-bdbb-a37e6186bb1d">

**Comment by reiniervlinschoten** [comment]
@martijnkersloot This will be fixed in v2.0.0, but I see a distinct statement in this piece of code:
https://github.com/castoredc/castoRedc/blob/ff4aeaf203360bf22a3988b3bfff837d6cd476bd/R/CastorData.R#L248-L255

I cannot figure out why that is there, as there should be no duplicates in this piece of data
```r
select(repeating_data_instances, participant_id, field_id, repeating_data_name,
                   created_on, repeating_data_instance_name, field_value)
```

And it could also not be a hotfix for the bug described in this issue, as the created_on times would be different and distinct would not work.

**Comment by slknijnenburg** [comment]
@reiniervlinschoten I wouldn't be surprised if the `distinct()` call was a non-working attempt at a fix for the issue. I can't imagine any other good reasons why to use distinct here, and I think (🤞) that it can be removed.
